packages/esm-messages-app/src/messages/messages-chart-dashboard.component.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useCallback, useMemo } from 'react';
+import classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
+import {
+  Button,
+  DataTable,
+  DataTableSkeleton,
+  InlineLoading,
+  Layer,
+  OverflowMenu,
+  OverflowMenuItem,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+  Tile,
+} from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import { useConfig, useLayoutType } from '@openmrs/esm-framework';
+import { CardHeader, EmptyState, ErrorState, launchPatientWorkspace } from '@openmrs/esm-patient-common-lib';
+import { useMessagesTemplates } from '../hooks/useMessagesTemplates';
+import styles from './messages-chart-dashboard.scss';
+import { type MessagesTemplate } from '../types';
+import { type ConfigObject } from '../config-schema';
+
+function MessagesChartDashboard() {
+  const { t } = useTranslation();
+  const displayText = t('messages', 'Messages');
+  const headerTitle = t('messages', 'Messages');
+  const { endOfMessageType } = useConfig<ConfigObject>();
+  const layout = useLayoutType();
+  const isTablet = layout === 'tablet';
+  const isDesktop = layout === 'small-desktop' || layout === 'large-desktop';
+
+  const { messagesTemplates, isLoadingTemplates, error, isValidatingTemplates, mutateTemplates } =
+    useMessagesTemplates();
+
+  const headers = useMemo(
+    () => [
+      {
+        key: 'name',
+        header: t('messageType', 'Message Type'),
+        isSortable: true,
+        sortFunc: (valueA, valueB) => valueA.display?.localeCompare(valueB.display),
+      },
+      {
+        key: 'status',
+        header: t('status', 'Status'),
+        isSortable: true,
+        sortFunc: (valueA, valueB) => valueA.display?.localeCompare(valueB.display),
+      },
+    ],
+    [t],
+  );
+
+  const getStatus = useCallback(
+    (template: MessagesTemplate) => {
+      const defaultValue = template.templateFields.find((field) => field.type === 'SERVICE_TYPE')?.defaultValue ?? null;
+
+      if (defaultValue) {
+        const endOfMessages =
+          template.templateFields.find((field) => field.type === 'END_OF_MESSAGES')?.defaultValue ?? null;
+        const endOfMessageValue = endOfMessages !== 'NO_DATE|EMPTY' ? endOfMessages.split('|')[1] : '';
+        return defaultValue === 'Deactivate service'
+          ? t('deactivated', 'DEACTIVATED')
+          : `${defaultValue}, Started: ${new Date(template.createdAt).toISOString().split('T')[0]}, Ends: ${t('endOfMessagesType', 'After {{endOfMessageValue}} times', { endOfMessageValue })}`;
+      } else {
+        return '';
+      }
+    },
+    [t],
+  );
+
+  const tableRows = useMemo(() => {
+    if (!messagesTemplates.length) return [];
+    return messagesTemplates?.map((template) => {
+      return {
+        ...template,
+        id: template.uuid,
+        status: getStatus(template),
+      };
+    });
+  }, [getStatus, messagesTemplates]);
```

Comment:
Not important, but it is possible to write this this way.

```suggestion
  const tableRows = useMemo(() =>
   messageTemplates?.map((template) => ({
     ...template,
     id: template.uuid
     status: getStatus(template)
   }) ?? []
   , [getStatus, messagesTemplates]);
```